### PR TITLE
Not fetching the blueprint of installations if not needed

### PR DESCRIPTION
**Pull request by achimweigel**
**How to categorize this PR?**

/area control-plane
/kind bug
/priority 3

**What this PR does / why we need it**:

A root installation fails if the resolution of the blueprint of a sibling installation fails. Therefore we reduced the cases in which the blueprint is resolved to those where it is really needed. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

We tested the fix against the failing landscaper installation for the dev landscape and the problems were resolved.

**Release note**:

```bugfix operator
- root installation do not fail anymore if a sibling root installation fails to resolve its blueprint
```

### CODE REVIEW

File: pkg/landscaper/installations/installation.go [hunk 1]
```diff
@@ -7,27 +7,35 @@ package installations
...
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
 	"github.com/gardener/landscaper/pkg/landscaper/blueprints"
 )
 
-// Installation is the internal representation of a installation
-type Installation struct {
-	Imports   map[string]interface{}
-	Info      *lsv1alpha1.Installation
-	Blueprint *blueprints.Blueprint
+type InstallationBaseInterface interface {
```
Comment (schrodit) [inline pkg/landscaper/installations/installation.go:16]: not sure if we really need the interface.
I think just having the `InstallationBase` struct is enough.

**Review by schrodit — CHANGES_REQUESTED** [review]

File: pkg/landscaper/installations/installation.go (hunk 1 above)
Comment (achimweigel) [reply pkg/landscaper/installations/installation.go:16]: We introduced the interface to be able to provide Installations and InstallationBases to methods which only need objects fulfilling the contract of InstallationBaseInterface. Of course we could go another way by removing the interface and changing the contract of those functions to InstallationBase, but then we must convert Installations first into InstallationBases when these are called.

Comment (achimweigel) [reply pkg/landscaper/installations/installation.go]: we removed the interface

File: pkg/landscaper/installations/imports/validation_test.go [hunk 2]
```diff
@@ -141,16 +141,16 @@ var _ = Describe("Validation", func() {
...
 		Expect(err).ToNot(HaveOccurred())
-		inInstA.Info.Status.Phase = lsv1alpha1.ComponentPhaseProgressing
-		Expect(fakeClient.Update(ctx, inInstA.Info)).To(Succeed())
+		inInstA.GetInfo().Status.Phase = lsv1alpha1.ComponentPhaseProgressing
+		Expect(fakeClient.Update(ctx, inInstA.GetInfo())).To(Succeed())
 
 		inInstB, err := installations.CreateInternalInstallation(ctx, op, fakeInstallations["test1/b"])
 		Expect(err).ToNot(HaveOccurred())
-		inInstB.Info.Status.Phase = lsv1alpha1.ComponentPhaseSucceeded
+		inInstB.GetInfo().Status.Phase = lsv1alpha1.ComponentPhaseSucceeded
```
Comment (schrodit) [inline pkg/landscaper/installations/imports/validation_test.go:149]: As we moved to the struct approach I think we do not need that getter indirection anymore.
So we could just leave most of the code as it is
```suggestion
		inInstB.Info.Status.Phase = lsv1alpha1.ComponentPhaseSucceeded
```

File: pkg/landscaper/installations/installation.go [hunk 3]
```diff
@@ -107,6 +96,42 @@ func (i *Installation) IsImportingTarget(name string) bool {
...
 
 // MergeConditions updates or adds the given condition to the installation's condition.
-func (i *Installation) MergeConditions(conditions ...lsv1alpha1.Condition) {
+func (i *InstallationBase) MergeConditions(conditions ...lsv1alpha1.Condition) {
 	i.Info.Status.Conditions = lsv1alpha1helper.MergeConditions(i.Info.Status.Conditions, conditions...)
 }
+
+// Installation is the internal representation of a installation
+type Installation struct {
+	*InstallationBase
```
Comment (schrodit) [inline pkg/landscaper/installations/installation.go:105]: why is the pointer needed?
I think a pointer here makes only sense if both structs could live on their own in parallel.
Or if the installation base could be nil but that could lead to more problems.

I propose that we just embed the struct.
```suggestion
	InstallationBase
```

**Review by schrodit — CHANGES_REQUESTED** [review]

**Comment by schrodit** [comment]
close in favor of https://github.com/gardener/landscaper/pull/197

**Pull request closed without merging**